server/src/main/java/org/apache/cloudstack/agent/lb/IndirectAgentLBServiceImpl.java

```diff
@@ -136,7 +136,7 @@ public int compare(Long x, Long y) {
         }
         final List <Host> agentBasedHosts = new ArrayList<>();
         for (final Host host : allHosts) {
-            if (host == null || host.getResourceState() != ResourceState.Enabled) {
+            if (host == null || host.getResourceState() == ResourceState.Creating || host.getResourceState() == ResourceState.Error) {
```

Comment:
Probably I am missing something here. @nvazquez Why not considering also `host.getResourceState() ==  ResourceState.PrepareForMaintenance`, and `host.getResourceState() ==  ResourceState.ErrorInMaintenance`?

Reply:
The idea is that even in those states, we still send an updated list of hosts and/or the algorithm - we want to make sure that almost EVERY host's state will be OK to receive the updated list. (why not?)

Reply:
Makes sense @andrijapanicsb @nvazquez.

Reply:
Hi @GabrielBrascher sorry for the late response Yes, as Andrija mentions those states are considered as well. The table added is great to show multiple use cases, thanks @andrijapanicsb